File: app/src/test/java/me/shreyasr/chatse/TestUtils.java
package me.shreyasr.chatse;

import android.content.SharedPreferences;

import org.mockito.Mockito;

import java.util.Set;

import static org.mockito.Mockito.mock;

public class TestUtils {

    public static SharedPreferences getMockPrefs() {
        SharedPreferences prefs = mock(SharedPreferences.class);
        SharedPreferences.Editor editor = new SharedPreferences.Editor() {
            @Override public SharedPreferences.Editor putString(String key, String value) { return this; }
            @Override public SharedPreferences.Editor putStringSet(String key, Set<String> values) { return this; }
            @Override public SharedPreferences.Editor putInt(String key, int value) { return this; }
            @Override public SharedPreferences.Editor putLong(String key, long value) { return this; }
            @Override public SharedPreferences.Editor putFloat(String key, float value) { return this; }
            @Override public SharedPreferences.Editor putBoolean(String key, boolean value) { return this; }
            @Override public SharedPreferences.Editor remove(String key) { return this; }
            @Override public SharedPreferences.Editor clear() { return this; }
            @Override public boolean commit() { return false; }
            @Override public void apply() { }
        };
        Mockito.doReturn(editor).when(prefs).edit();
        return prefs;
    }
}

File: app/src/test/java/com/tristanwiley/chatse/TestUtils.kt
package com.tristanwiley.chatse

import android.content.SharedPreferences
import com.nhaarman.mockitokotlin2.doReturn
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever

fun getMockPrefs(): SharedPreferences {
    val prefs = mock<SharedPreferences>()
    val editor = object : SharedPreferences.Editor {
        override fun clear() = this

        override fun putLong(key: String?, value: Long) = this

        override fun putInt(key: String?, value: Int) = this

        override fun remove(key: String?) = this

        override fun putBoolean(key: String?, value: Boolean) = this

        override fun putStringSet(key: String?, values: MutableSet<String>?) = this

        override fun commit() = false

        override fun putFloat(key: String?, value: Float) = this

        override fun apply() = Unit

        override fun putString(key: String?, value: String?) = this

    }

    whenever(prefs.edit()).doReturn(editor)
    return prefs
}
